src/components/audio-params.js

```diff
@@ -0,0 +1,271 @@
+      this.updateAttenuation();
+    }
+
+    if (this.normalizer !== null) {
+      this.normalizer.apply();
+    } else {
+      // We one only enable the Normalizer for avatar-audio-source components
+      if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+        this.enableNormalizer();
+      }
+    }
+  },
+
+  audio() {
+    if (this.data.sourceType === SourceType.AVATAR_RIG) {
+      // Create fake parametes for the avatar rig as it doens't have an audio source.
+      const audioParams = this.el.sceneEl.systems["hubs-systems"].audioSettingsSystem.audioSettings;
+      const avatarRigObj = document.getElementById("avatar-rig").querySelector(".camera").object3D;
+      avatarRigObj.updateMatrixWorld(true);
+      const position = new THREE.Vector3();
+      avatarRigObj.getWorldPosition(position);
+      const direction = new THREE.Vector3(0, 0, -1);
+      const worldQuat = new THREE.Quaternion();
+      avatarRigObj.getWorldQuaternion(worldQuat);
+      direction.applyQuaternion(worldQuat);
+      return {
+        panner: {
+          orientationX: {
+            value: direction.x
+          },
+          orientationY: {
+            value: direction.y
+          },
+          orientationZ: {
+            value: direction.z
+          },
+          positionX: {
+            value: position.x
+          },
+          positionY: {
+            value: position.y
+          },
+          positionZ: {
+            value: position.z
+          },
+          distanceModel: audioParams.avatarDistanceModel,
+          maxDistance: audioParams.avatarMaxDistance,
+          refDistance: audioParams.avatarRefDistance,
+          rolloffFactor: audioParams.avatarRolloffFactor,
+          coneInnerAngle: audioParams.avatarConeInnerAngle,
+          coneOuterAngle: audioParams.avatarConeOuterAngle,
+          coneOuterGain: audioParams.avatarConeOuterGain
+        }
+      };
+    } else if (this.data.sourceType === SourceType.MEDIA_VIDEO) {
+      const audio = this.el.getObject3D("sound");
+      return audio instanceof THREE.PositionalAudio ? audio : null;
+    } else if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+      const audio = this.el.getObject3D("avatar-audio-source");
+      return audio instanceof THREE.PositionalAudio ? audio : null;
+    }
+  },
+
+  enableNormalizer() {
+    const audio = this.audio();
+    if (audio) {
+      const avatarAudioSource = this.el.components["avatar-audio-source"];
+      if (avatarAudioSource) {
+        this.normalizer = new AudioNormalizer(audio);
+        avatarAudioSource.el.addEventListener("sound-source-set", () => {
+          const audio = avatarAudioSource && avatarAudioSource.el.getObject3D(avatarAudioSource.attrName);
+          if (audio) {
+            this.normalizer = new AudioNormalizer(audio);
+            this.normalizer.apply();
+          }
+        });
+      }
+    }
+  },
+
+  updateDistances() {
+    const listenerPos = new THREE.Vector3();
```

Comment:
True, moved to a component variable to create at init only.

Reply:
For temporary state light this I usually use a closure or a variable with the `tmp` prefix just defined outside of the component so that its not confused as some piece of state we actually care about keeping around over time. Pretty minor though.